### DISCUSSION TITLE
Using local topography files instead of through published containers

### DISCUSSION THREAD

**Discussion by sitangshugk95** [ContactEngineering/SurfaceTopography#331] [Q&A] [answered]
Are there any example files that show how to use local topography files (AFM scans/XYZ data/MATLAb files) instead of read_published_container?

**Comment by pastewka** [comment]
The container is a collection (list) of measurements, if you want to read individual files you can just do that with `read_topography`, i.e. `from SurfaceTopography import read_topography`. The reader autodetects the format if supported. You should be able to directly read the data file that you AFM produces.

**Comment by pastewka** [comment, marked as answer]
Note that if you read MATLAB files, you need to provide some additional information to the reader, in particular the size of the measurement and the unit. You can do this via `read_topography('myfile.mat', physical_sizes=(1, 2), unit='mm')` (of course adjust the numbers appropriately).

**Comment by pastewka** [comment]
There is limited documentation here:

https://contactengineering.github.io/SurfaceTopography/usage.html

https://contactengineering.github.io/SurfaceTopography/source/SurfaceTopography.IO.html#SurfaceTopography.IO.read_topography
